src/Twig/Extension/PageExtension.php

```diff
@@ -183,15 +189,21 @@ public function renderContainer(string $name, $page = null, array $options = [])
      */
     public function renderBlock(PageBlockInterface $block, array $options = []): string
     {
-        if (
-            false === $block->getEnabled()
-            && !$this->cmsManagerSelector->isEditor()
-            && $this->hideDisabledBlocks
-        ) {
+        try {
+            if (
+                false === $block->getEnabled()
+                && false === $this->cmsManagerSelector->isEditor()
+                && $this->hideDisabledBlocks
+            ) {
+                return '';
+            }
+
+            return $this->blockHelper->render($block, $options);
+        } catch (BlockNotFoundException $exception) {
```

Comment:
Why BlockNotFoundException ?

I thought we introduced a BlockServiceNotFoundException ; you need to catch something else ?

Reply:
Yeah it's the old exception I need to change!